bitmovin/resources/models/codecconfigurations/av1_codec_configuration.py

```diff
@@ -0,0 +1,275 @@
+        self.clientInitialBufferSize = client_initial_buffer_size
+        self.clientOptimalBufferSize = client_optimal_buffer_size
+        self.tileColumns = tile_columns
+        self.tileRows = tile_rows
+        self.automaticAltRefFramesEnabled = automatic_alt_ref_frames_enabled
+        self.arnrMaxFrames = arnr_max_frames
+        self.arnrStrength = arnr_strength
+        self.cqLevel = cq_level
+        self.maxIntraRate = max_intra_rate
+        self.maxInterRate = max_inter_rate
+        self.gfCbrBoost = gf_cbr_boost
+        self.lossless = lossless
+        self.frameParallel = frame_parallel
+        self.sharpness = sharpness
+        self.frameBoostEnabled = frame_boost_enabled
+        self.noiseSensitivity = noise_sensitivity
+        self.minGfInterval = min_gf_interval
+        self.maxGfInterval = max_gf_interval
+        self.numTileGroups = num_tile_groups
+        self.mtuSize = mtu_size
+
+    @property
+    def colorConfig(self):
+        return self._colorConfig
+
+    @colorConfig.setter
+    def colorConfig(self, new_color_config):
+        if new_color_config is None:
+            self._colorConfig = None
+        elif isinstance(new_color_config, ColorConfig):
+            self._colorConfig = new_color_config
+        else:
+            raise InvalidTypeError('colorConfig has to be of type ColorConfig')
+
+    @property
+    def keyPlacementMode(self):
+        return self._keyPlacementMode
+
+    @keyPlacementMode.setter
+    def keyPlacementMode(self, new_key_placement_mode):
+        if new_key_placement_mode is None:
+            return
+        if isinstance(new_key_placement_mode, str):
+            self._keyPlacementMode = new_key_placement_mode
+        elif isinstance(new_key_placement_mode, AV1KeyPlacementMode):
+            self._keyPlacementMode = new_key_placement_mode.value
+        else:
+            raise InvalidTypeError('Invalid type {} for keyPlacementMode: must be either str or AV1KeyPlacementMode.'
+                                   .format(type(new_key_placement_mode)))
+
+    @property
+    def rateControlMode(self):
+        return self._rateControlMode
+
+    @rateControlMode.setter
+    def rateControlMode(self, new_rate_control_mode):
+        if new_rate_control_mode is None:
+            return
+        if isinstance(new_rate_control_mode, str):
+            self._rateControlMode = new_rate_control_mode
+        elif isinstance(new_rate_control_mode, AV1RateControlMode):
+            self._rateControlMode = new_rate_control_mode.value
+        else:
+            raise InvalidTypeError('Invalid type {} for rateControlMode: must be either str or AV1RateControlMode.'
+                                   .format(type(new_rate_control_mode)))
+
+    @property
+    def adaptiveQuantMode(self):
+        return self._adaptiveQuantMode
+
+    @adaptiveQuantMode.setter
+    def adaptiveQuantMode(self, new_adaptive_quant_mode):
+        if new_adaptive_quant_mode is None:
```

Comment:
Shouldn't `self._adaptiveQuantMode` set to `None` here?

Reply:
I see mixed patterns across the codec configurations, some just return and others set the private property. Makes sense to set the private property in the settor, so will update.